File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.swift
import Foundation

func solution(_ code:String) -> String {
    var mode: Bool = false
    var ret: String = ""
    
    for (index, char) in code.enumerated() {
        guard char != "1" else {
            mode.toggle() 
            continue
        }
        
        if mode {
            if index % 2 == 1 { ret.append(char) }
        } else {
            if index % 2 == 0 { ret.append(char) }
        }
    }
    
    return ret.isEmpty ? "EMPTY" : ret
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.cs
using System;

public class Solution {
    public string solution(string code) {
        bool mode = false;
        string ret = "";
        
        for (int i = 0; i < code.Length; i++)
        {
            if (code[i] == '1') {
                mode = !mode;
                continue;
            }
            
            if (mode && (i % 2 == 1))
            {
                ret += code[i];
            }
            else if (mode == false && i % 2 == 0)
            {
                ret += code[i];
            }
        }
        
        return String.IsNullOrEmpty(ret) ? "EMPTY" : ret;
    }
}